tests/autobahn/test_autobahn.py

```diff
@@ -0,0 +1,163 @@
+import errno
+import json
+import shutil
+import socket
+import subprocess
+import sys
+import time
+from pathlib import Path
+from typing import Any, Dict, List
+
+import pytest
+
+HOST = "localhost"
+PORT = 9001
+
+
+@pytest.fixture(scope="module", autouse=True)
+def create_report_directory(request: Any) -> None:
+    path = Path(f"{request.fspath.dirname}/reports")
+    if path.is_dir():
+        shutil.rmtree(path)
+    path.mkdir()
+
+
+@pytest.fixture(scope="session", autouse=True)
+def setup_venv() -> None:
+    if not Path("tests/autobahn/autobahntestsuite-env").exists():
+        print("Creating Python 2.7 environment and installing autobahntestsuite")
+        subprocess.check_call(
+            ["virtualenv", "-p", "python2.7", "tests/autobahn/autobahntestsuite-env"]
+        )
+        subprocess.check_call(
+            [
+                "tests/autobahn/autobahntestsuite-env/bin/pip",
+                "install",
+                "autobahntestsuite>=0.8.0",
+            ]
+        )
+
+
+def wait_for_server(host: str, port: int) -> None:
+    while True:
+        sock = socket.socket()
+        try:
+            sock.connect((HOST, PORT))
+        except OSError as sock_err:
+            if sock_err.errno == errno.ECONNREFUSED:
+                time.sleep(0.01)
+            else:
+                raise
```

Comment:
Would this be cleaner?
```suggestion
        except ConnectionRefusedError:
            time.sleep(0.01)
```
https://docs.python.org/3/library/exceptions.html#ConnectionRefusedError